src/main/MQ2ImGuiConsole.cpp

```diff
@@ -27,1504 +27,1638 @@
 
 #pragma endregion
 
-//============================================================================
+	//============================================================================
 
 #pragma region ImGui Console
 
-class ImGuiConsole
-{
-public:
-	char m_inputBuffer[2048];
-	ImVector<const char*> m_commands;
-	std::vector<std::string> m_history;
-	int m_historyPos = -1;    // -1: new line, 0..History.Size-1 browsing history.
-	bool m_scrollToBottom = true;
-	std::unique_ptr<ImGuiZepConsole> m_zepEditor;
-
-	ImGuiConsole()
-	{
-		ZeroMemory(m_inputBuffer, lengthof(m_inputBuffer));
-		m_zepEditor = std::make_unique<ImGuiZepConsole>();
-	}
-
-	~ImGuiConsole()
-	{
-		ClearLog();
-	}
-
-	void ClearLog()
-	{
-		m_zepEditor->Clear();
-	}
-
-	template <typename... Args>
-	void AddLog(ImU32 color, std::string_view fmt, const Args&... args)
+	class ImGuiConsole
 	{
-		fmt::basic_memory_buffer<char> buf;
-		fmt::format_to(fmt::appender(buf), fmt, args...);
+	public:
+		char m_inputBuffer[2048];
+		ImVector<const char*> m_commands;
+		std::vector<std::string> m_history;
+		int m_historyPos = -1;    // -1: new line, 0..History.Size-1 browsing history.
+		bool m_scrollToBottom = true;
+		std::chrono::steady_clock::time_point m_pr_StartTime;
+		std::unique_ptr<ImGuiZepConsole> m_zepEditor;
+
+		ImGuiConsole()
+		{
+			ZeroMemory(m_inputBuffer, lengthof(m_inputBuffer));
+			m_zepEditor = std::make_unique<ImGuiZepConsole>();
+		}
 
-		m_zepEditor->AppendFormattedText(std::string_view(buf.data(), buf.size()), color, false);
-	}
+		~ImGuiConsole()
+		{
+			ClearLog();
+		}
 
-	template <typename... Args>
-	void AddLog(std::string_view fmt, const Args&... args)
-	{
-		AddLog(s_defaultColor, std::move(fmt), args...);
-	}
+		void ClearLog()
+		{
+			m_zepEditor->Clear();
+		}
 
-	void AddWriteChatColorLog(const char* line, ImU32 defaultColor = s_defaultColor, bool newline = false)
-	{
-		m_zepEditor->AppendFormattedText(line, defaultColor, newline);
-	}
+		template <typename... Args>
+		void AddLog(ImU32 color, std::string_view fmt, const Args&... args)
+		{
+			fmt::basic_memory_buffer<char> buf;
+			fmt::format_to(fmt::appender(buf), fmt, args...);
 
-	void Draw(bool* pOpen)
-	{
-		ImGuiWindowFlags windowFlags = ImGuiWindowFlags_MenuBar;
+			m_zepEditor->AppendFormattedText(std::string_view(buf.data(), buf.size()), color, false);
+		}
 
-		ImGui::SetNextWindowSize(ImVec2(640, 240), ImGuiCond_FirstUseEver);
-		ImGui::PushStyleVar(ImGuiStyleVar_WindowPadding, ImVec2(1, 0));
+		template <typename... Args>
+		void AddLog(std::string_view fmt, const Args&... args)
+		{
+			AddLog(s_defaultColor, std::move(fmt), args...);
+		}
 
-		if (!ImGui::Begin("MacroQuest Console", pOpen, windowFlags))
+		void AddWriteChatColorLog(const char* line, ImU32 defaultColor = s_defaultColor, bool newline = false)
 		{
-			ImGui::End();
+			m_zepEditor->AppendFormattedText(line, defaultColor, newline);
+		}
 
-			ImGui::PopStyleVar();
-			return;
+		ImGuiWindowFlags GetConsolesFlags()
```

Comment:
this doesn't really need to be a function, but theres a typo in its name anyhow.